### ISSUE TITLE
求问：输入一致，在pycharm  debug下，输出结果不一样

### ISSUE THREAD

**Issue by Aussi12** [PaddlePaddle/Paddle#53524]
### 请提出你的问题 Please ask your question

相同模型，相同输入  在pycharm  debug下，输出结果却不一样为什么呢

**Comment by risemeup1** [comment]
你能写的具体点吗？方便我们这边定位

**Comment by Aussi12** [comment]
目前正在复现模型trl，里面生成的结果，Q+R 再次输入到模型获取logits。 debug代码段如下
input_kwargs = {key: paddle.Tensor(np.array(value[i * fbs : (i + 1) * fbs])) for key, value in model_inputs.items()}
query_batch = queries[i * fbs : (i + 1) * fbs]
response_batch = responses[i * fbs : (i + 1) * fbs]
input_kwargs['value_flag']=True
logits,values = model(**input_kwargs)

最后一行在debug的pycharm环境下，输入三次获得的logits不一样，我这个代码是有点bug，但是这里不应该不一样的，请求解决

**Comment by Aussi12** [comment]
> 你能写的具体点吗？方方便我们这边定位置
目标之前正确在恢复模型trl，里面生成的结果，Q+R再次输入到模型获得logits。 debug代码段如下
input_kwargs = {key: paddle.Tensor(np.array(value[i * fbs : (i + 1) * fbs])) for key, value in model_inputs.items()}
query_batch = queries[i * fbs : (i + 1) * fbs]
response_batch = responses[i * fbs : (i + 1) * fbs]
input_kwargs[ 'value_flag']=True
logits，values = model(**input_kwargs)

最后一步在debug的pycharm环境下，输入三次获得的logits不一样，我这个代码是有点bug，但是这里不应该不一样的，请求解决

**Comment by Aussi12** [comment]
> 你能写的具体点吗？方便我们这边定位

代码已经重新调试，但仍存在不一致的情况。请求解决🙏

**Comment by risemeup1** [comment]
你好，逐步怀疑有两种可能
1 你可以在每次debug得时候看下参数**input_kwargs是不是一样的
2 model()本身是不是带有随机性

**Comment by Aussi12** [comment]
> 你好，逐步怀疑有两种可能 1你可以在每次debug得时候看下参数**input_kwargs是不是一样的 2 model()本身是不是带随机性

两种可能都不存在
条件一在同一个debug的cons端口中
条件二，模型输出的的logit概率，所有参数，原理是应该保持一致的

**Comment by risemeup1** [comment]
请尽量提供完整的代码片段吧，可能都不是paddle本身的问题

**Comment by Aussi12** [comment]
> 请尽可能提供完整的代码片段吧，可能都不是桨本身的问题

[trl_reanforce.tar.gz](https://github.com/PaddlePaddle/Paddle/files/11411689/trl_reanforce.tar.gz)
代码文件如上

**Comment by risemeup1** [comment]
麻烦你把这个diff发出来，是很大吗？

**Comment by Aussi12** [comment]
> 麻烦你把这个diff拿出来，很大吗？

没懂 啥意思呀

**Comment by risemeup1** [comment]
> > 你能写的具体点吗？方方便我们这边定位置
> > 目标之前正确在恢复模型trl，里面生成的结果，Q+R再次输入到模型获得logits。 debug代码段如下
> > input_kwargs = {key: paddle.Tensor(np.array(value[i * fbs : (i + 1) * fbs])) for key, value in model_inputs.items()}
> > query_batch = queries[i * fbs : (i + 1) * fbs]
> > response_batch = responses[i * fbs : (i + 1) * fbs]
> > input_kwargs[ 'value_flag']=True
> > logits，values = model(**input_kwargs)
> 
> 最后一步在debug的pycharm环境下，输入三次获得的logits不一样，我这个代码是有点bug，但是这里不应该不一样的，请求解决

三次获得logits差别很大吗？

**Comment by Aussi12** [comment]
> 三次获得logits差别很大吗？

0.1，感觉还是挺大的。目前还没时间仔细看，只是用用，里边的损失函数比较复杂。

**Comment by risemeup1** [comment]
> > 三次获得logits差别很大吗？
> 
> 0.1，感觉还是挺大的。目前还没时间仔细看，只是用用，里边的损失函数比较复杂。

如果差别比较小的话会是Paddle的问题，差别挺大的话，应该不是Paddle的问题
